### PR TITLE
Adding the SPGM separation model for WSJ0-2Mix recipe

### PR DESCRIPTION
**Pull request by Yip-Jia-Qi**
## What does this PR do?
This PR adds support for a new model SPGM, that improves upon sepformer by replacing the interblock with a Global Pooling and Modulation layer.

SPGM was recently accepted to ICASSP 2024. https://arxiv.org/abs/2309.12608 

The WSJ0-2Mix model and weights are released and the config file is provided. The model is added to the speechbrain/lobes/model directory.

This PR does not introduce any breaking changes.

<details>
  <summary><b>Before submitting</b></summary>

- [x] Did you read the [contributor guideline](https://speechbrain.readthedocs.io/en/latest/contributing.html)?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing [tests](https://github.com/speechbrain/speechbrain/tree/develop/tests) pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [x] Does your code adhere to project-specific code style and conventions?

</details>

## PR review

<details>
  <summary>Reviewer checklist</summary>

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified
- [ ] Confirm that the changes adhere to compatibility requirements (e.g., Python version, platform)
- [ ] Review the self-review checklist to ensure the code is ready for review

</details>

### CODE REVIEW

**Comment by ycemsubakan** [comment]
Thanks a lot for your commit @Yip-Jia-Qi , I will review the PR soon.

**Comment by ycemsubakan** [comment]
Thanks @Yip-Jia-Qi  for your contribution. Overall the addition looks good to me. I noticed one missing docstring in `SPGM.py` (see the inline comment above) 

Another thing that is missing is the checkpoints for the best model you have trained. Please see inline comment above also. 

Here are some of my modifications:
- I changed the PR title to make it more self-descriptive. 
- I fixed the other failing docstring tests.  
- I am noticing that the tests still fail after the changes, but it's not because of the changes introduced in this PR. @mravanelli any idea why this might be happening?  

I am also running the model to see if I can reproduce your numbers. With 5 epochs here's what I have (without DM). Is this similar to what you have? I will run more to also confirm that we converge to the claimed numbers. 
```
epoch: 1, lr: 1.50e-04 - train si-snr: -6.80e+00 - valid si-snr: -9.03e+00
epoch: 2, lr: 1.50e-04 - train si-snr: -1.08e+01 - valid si-snr: -1.18e+01
epoch: 3, lr: 1.50e-04 - train si-snr: -1.30e+01 - valid si-snr: -1.32e+01
epoch: 4, lr: 1.50e-04 - train si-snr: -1.42e+01 - valid si-snr: -1.43e+01
epoch: 5, lr: 1.50e-04 - train si-snr: -1.51e+01 - valid si-snr: -1.53e+01
```

**Comment by Yip-Jia-Qi** [comment]
> Thanks @Yip-Jia-Qi for your contribution. Overall the addition looks good to me. I noticed one missing docstring in `SPGM.py` (see the inline comment above)
> 
> Another thing that is missing is the checkpoints for the best model you have trained. Please see inline comment above also.
> 
> Here are some of my modifications:
> 
> * I changed the PR title to make it more self-descriptive.
> * I fixed the other failing docstring tests.
> * I am noticing that the tests still fail after the changes, but it's not because of the changes introduced in this PR. @mravanelli any idea why this might be happening?
> 
> I am also running the model to see if I can reproduce your numbers. With 5 epochs here's what I have (without DM). Is this similar to what you have? I will run more to also confirm that we converge to the claimed numbers.
> 
> ```
> epoch: 1, lr: 1.50e-04 - train si-snr: -6.80e+00 - valid si-snr: -9.03e+00
> epoch: 2, lr: 1.50e-04 - train si-snr: -1.08e+01 - valid si-snr: -1.18e+01
> epoch: 3, lr: 1.50e-04 - train si-snr: -1.30e+01 - valid si-snr: -1.32e+01
> epoch: 4, lr: 1.50e-04 - train si-snr: -1.42e+01 - valid si-snr: -1.43e+01
> epoch: 5, lr: 1.50e-04 - train si-snr: -1.51e+01 - valid si-snr: -1.53e+01
> ```

The output folder can be found here on my personal onedrive: https://1drv.ms/f/s!AlNG26WLrzhkiO91f48NCHrA2EVs7A?e=XXRlZL 
This is for the model without dynamic mixing reported in the paper. However, the model hyperparameters in this folder are not compatible with this PR because I've made some changes to the model's code to be neater for the PR, although the checkpoints should still load.
Here are my first 5 epochs for SPGM without dynamic mixing

> ```
>epoch: 1, lr: 1.50e-04 - train si-snr: -6.46e+00 - valid si-snr: -8.58e+00
>epoch: 2, lr: 1.50e-04 - train si-snr: -1.03e+01 - valid si-snr: -1.12e+01
>epoch: 3, lr: 1.50e-04 - train si-snr: -1.24e+01 - valid si-snr: -1.26e+01
>epoch: 4, lr: 1.50e-04 - train si-snr: -1.37e+01 - valid si-snr: -1.41e+01
>epoch: 5, lr: 1.50e-04 - train si-snr: -1.45e+01 - valid si-snr: -1.44e+01
> ```

The weights for the SPGM with dynamic mixing (which is the best model I've trained) is found on huggingface which i've added to the PR. I can upload the output folder as well.

Can I check that the links to the output folder of the training should be added to the readme?
